src/services/cloudwatchLogs/schema.graphql

```diff
@@ -3,6 +3,7 @@ type awsCloudwatchLog @key(fields: "arn") {
   accountId: String! @search(by: [hash])
   arn: String! @id @search(by: [hash, regexp])
   region: String @search(by: [hash, regexp])
+  cgId: String @search(by: [hash, regexp])
```

Comment:
why do we need to add this?

Reply:
Because we're trying to standardize the common attributes that an AWS schema could have. Notice that is nullable because for this service the cgId doesn't perform a role. To keep the solution agnostic at the RulesEngine, we're using the extraFields field to create connections, take a look [here](https://github.com/cloudgraphdev/sdk/blob/ab75afa8bfeb0949e053bb38440cae5ce1628ded/src/rules-engine/data-processors/dgraph-data-processor.ts#L172)

Reply:
but if its a standard field would it not be in the base schema and not here? I am just wondering why it is being added here but not to other schemas.

Reply:
how would this change the 'data contract'? At EP, we assume that every resource has a unique `id`. is it not possible to keep that nice property? (we'll need to code special cases to deal with this too)

Reply:
@aacotroneo sadly the reason is due mainly to cloudtrail. because cloudtrail can live in multiple accounts (but still has the arn of the account where it was created), our multi account scanning had an issue. Either create a new id (which we did which is cgId) or only show the cloudtrail in the account it "belongs" to. (which I considered but was suggested from @ckoning that users will want to see it in both accounts as it would show in both in the management console). I am wondering if we want to reconsider this decision. Currently we use cgId because cloudtrail _does_ have an id and we thought it may be confusing to users if the id field for cloudtrail in CG was NOT the actual cloudtrail id. Its a difficult problem with no easy solutions (outside of composite ids which dgraph doesnt have). @ckoning now that this problem has had some time to sit, what do you think would be best here?

Reply:
That's fine.. but does that really require extending the contract making everything else a tiny bit more complex, or can we solve the issue within cloudtrail service alone?
As I understand it, our `id` is already an opaque string, and we can put whatever we want in there (sometimes it's an arn, sometimes it's something else, or even a random uid for nested objects). Why not set the id equal to that `cgId` you mentioned and have a new field be cloudTrail-Id? We could have named those `_id` so it has a more internal connotation, but that's what our ids are.

Reply:
I would say my concern is that, while the `id` could be an id or an arn, it always (outside the cloudtrail) corresponds to a value DIRECTLY from aws, which cloudtrail would break that pattern if we moved it to id. It certainly is an option but that would be the draw back. Nested objs I am not too concerned with because they arent queried by the id (using a get query).

Reply:
Yes, but that's an unneeded restriction, and it's harder to reason about as a user of CG:

contract 1:
``` 
id - an opaque string that uniquely identifies a resource across accounts
```

contract 2:
```
id - a value that corresponds directly to an aws value. when that value is not unique across accounts refer to cgID
cgID: an opaque string that uniquely represents a resource across accounts, only meaningful when IDs clash
```

Reply:
yes I am not a fan of the `cgId` paradigm. I would prefer to not have to use it. But idk if the trade off of losing the value of contract 2 (part 1) would be worth moving to contract 1. I preferred only having the cloudtrail show up in the account that owned it (although I understand the drawbacks of that as well). Moving the random string to the `id` also brings with it another issue. The cgId value is only initialized in the format function, but the connections functions deal with the raw data. So the connections files would have no access to the value needed to form connections to the cloudtrail service. I suppose we could initialize it in the data function... 🤔

Reply:
Lets chat on a huddle for this so we can discuss pros and cons.